### PR TITLE
Improve resilience of adding tenants by implementing commit and rollback mechanism (part 2)

### PR DESCRIPTION
**Pull request by redouan-rhazouani**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/repos/db/migrator.go [hunk 1]
```diff
@@ -150,24 +150,52 @@ func (m *Migrator) UpdateShardStatus(ctx context.Context, className, shardName,
 	return idx.updateShardStatus(ctx, shardName, targetStatus)
 }
 
-func (m *Migrator) AddPartitions(ctx context.Context, class *models.Class, names []string) error {
+// CreatePartitions creates new partitions and returns a commit func
+// that can be used to either commit or rollback the partitions
+func (m *Migrator) NewPartitions(ctx context.Context, class *models.Class, partitions []string) (commit func(success bool), err error) {
```
Comment (parkerduckworth) [inline adapters/repos/db/migrator.go:155]: Probably you want to update this comment with the correct method name

File: adapters/repos/db/migrator.go [hunk 2]
```diff
@@ -150,24 +150,52 @@ func (m *Migrator) UpdateShardStatus(ctx context.Context, className, shardName,
...
+		}
+	}()
+
+	for _, name := range partitions {
+		if shard := idx.shards.Load(name); shard != nil {
+			continue
+		}
+		shard, err := NewShard(ctx, nil, name, idx, class, idx.centralJobQueue)
+		if err != nil {
+			return nil, fmt.Errorf("cannot create partition: %w", err)
```
Comment (parkerduckworth) [inline adapters/repos/db/migrator.go:192]: Would be helpful to include the partition name in this error message

File: adapters/repos/db/migrator.go (hunk 1 above)
Comment (redouan-rhazouani) [reply adapters/repos/db/migrator.go]: Absolutely 👍

File: adapters/repos/db/migrator.go (hunk 2 above)
Comment (redouan-rhazouani) [reply adapters/repos/db/migrator.go]: Good point! Done

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**